src/v3/src/transformer/dataSchema/addPhoneValidation.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { IdxMessage } from '@okta/okta-auth-js';
+
+import countryCallingCodes from '../../../../util/countryCallingCodes';
+import {
+  FieldElement,
+  TransformStepFn,
+} from '../../types';
+import { loc } from '../../util';
+import { traverseLayout } from '../util';
+
+export const getUniqueCountryCodes = (): string[] => {
+  const uniqueCodes: string[] = [];
+  const excludedCountries = ['HM', 'BV', 'TF'];
+  Object.entries(countryCallingCodes).forEach(([country, code]) => {
+    if (!uniqueCodes.includes(code) && !excludedCountries.includes(country)) {
+      uniqueCodes.push(code);
+    }
+  });
+  return uniqueCodes;
+};
+
+export const isPhoneNumberSet = (phoneNumber?: string): boolean => {
+  if (!phoneNumber) {
+    return false;
+  }
+
+  const uniqueCodes: string[] = getUniqueCountryCodes();
+  const countryCode = uniqueCodes.find((code) => {
+    const endIndex = code.length + 1;
+    return phoneNumber.substring(1, endIndex) === code;
+  });
+  if (!countryCode) {
+    return true;
+  }
+  const [phoneWithCountryCode] = phoneNumber.split('x');
+  const phoneNumberOnly = phoneWithCountryCode.substring(countryCode.length + 1);
+  return !!phoneNumberOnly;
+};
```

Comment:
@lesterchoi-okta I know we discussed possibly just using the number of characters to determine if the validation message should be thrown. However, after trying that, I noticed for countries with a short country code, the `empty field` error message would display even though the field was not empty. 

So I reverted back to this approach (assuming that invalid error would not be acceptable). 

Wanted to get your thoughts on either approach.

Reply:
possible to track country code from user interaction (component), then use it in validate fn?